sklearn/impute/_base.py

```diff
@@ -256,6 +256,14 @@ class SimpleImputer(_BaseImputer):
      [10.   3.5  9. ]]
     """
 
+    _parameter_constraints = {
+        **_BaseImputer._parameter_constraints,
+        "strategy": [StrOptions({"mean", "median", "most_frequent", "constant"})],
+        "fill_value": [str, numbers.Number, None],  # type: ignore
+        "verbose": ["verbose", Hidden(StrOptions({"deprecated"}))],  # type: ignore
+        "copy": ["boolean"],
+    }
```

Comment:
I think giving `_parameter_constraints` types is good enough to remove the mypy warnings:

```suggestion
    _parameter_constraints: dict = {
        **_BaseImputer._parameter_constraints,
        "strategy": [StrOptions({"mean", "median", "most_frequent", "constant"})],
        "fill_value": [str, numbers.Number, None],
        "verbose": ["verbose", Hidden(StrOptions({"deprecated"}))],
        "copy": ["boolean"],
    }
```